spec/lib/scrabble/player_spec.rb

```diff
@@ -0,0 +1,153 @@
+require "./lib/scrabble/player"
+
+describe "PLAYER" do
+
+  before :each do
+    @player = Scrabble::Player.new("George")
+  end
+
+  context "CAN HAS EXIST! (Player.new)" do
+    it "is an instance of Player." do
+      expect(@player).to be_an_instance_of(Scrabble::Player)
+    end
+
+    it "has a provided name." do
+      expect(@player.name).to eq("George")
+
+      new_player = Scrabble::Player.new("Larry Hotter")
+      expect(new_player.name).to eq("Larry Hotter")
+    end
+
+    it "has zero plays to start." do
+      expect(@player.plays.length).to eq(0)
+    end
+
+    it "as a result, hasn't won the game yet." do
+      expect(@player.won?).to eq(false)
+    end
+  end
+
+  context "CAN HAS TILES!" do
+    before :each do
+      @bag_o_tiles = Scrabble::TileBag.new
+    end
+
+    it "has the ability to draw tiles." do
+      @player.draw_tiles(@bag_o_tiles)
+
+      # actual test
+      expect(@player.tiles.length).to eq(7)
+    end
+
+    it "cannot draw past 7 tiles." do
+      # !Q how to prove?
```

Comment:
I couldn't think of a condition to prove that drawing past 7 was impossible aside from what I did with just calling it three times in a row.

~~Unnecessary aside: I think proofs were the reason I lost interest in higher level maths; I have a lot of trouble thinking of the opposite must-fail condition.~~